batch2/create-batch-tables.sql

```diff
@@ -196,6 +267,8 @@ BEGIN
 
     IF actual_n_jobs = expected_n_jobs THEN
       UPDATE batches SET closed = 1 WHERE id = in_batch_id;
+      UPDATE batches SET time_completed = UNIX_TIMESTAMP(NOW(3))
+        WHERE id = in_batch_id AND n_completed = batches.n_jobs;
```

Comment:
I would prefer the driver pass in the time rather than using getting the time in the database.  This is partially for consistency (of code), that the resolution/precision of the clocks may not be the same (you're not truncating Python time to 3 digits), and partially to rely on as few clocks as possible to minimize clock skew.

Reply:
ok.